tests/drivers/led/led_api/src/main.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2020 Seagate Technology LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+
+#include "test_led_api.h"
+
+void test_main(void)
+{
+	ztest_test_suite(led_test,
+			 ztest_unit_test(test_led_setup),
+			 ztest_unit_test(test_led_get_info),
+			 ztest_unit_test(test_led_on),
+			 ztest_unit_test(test_led_off),
+			 ztest_unit_test(test_led_set_color),
+			 ztest_unit_test(test_led_set_brightness));
+	ztest_run_test_suite(led_test);
+
+	/* User mode. */
+
```

Comment:
need to add `k_object_access_grant(device_get_binding(LED_CTRL_DEV_NAME), k_current_get());` here.

This will grant access permission for the LED device to the main thread. main thread runs in supervisor mode so it doesn't care, but its child user threads (the individual ztest cases) will inherit that permission, and if they don't have it they will generate a kernel oops due to insufficient access permission.

`k_object_access_grant()` is smart and will disregard bad output from `device_get_binding()`, there's a kernel-side hashtable of valid kernel object pointers it looks stuff up in before doing anything.